language/stdlib/modules/DesignatedDealer.move

```diff
@@ -241,25 +318,28 @@ module DesignatedDealer {
     /// that amount that can be minted according to the bounds for the `tier_index` tier.
     fun validate_and_record_mint<CoinType>(dd_addr: address, amount: u64, tier_index: u64)
     acquires TierInfo {
-        let tier = borrow_global_mut<TierInfo<CoinType>>(dd_addr);
-        reset_window(tier);
-        let cur_inflow = tier.window_inflow;
+        let tier_info = borrow_global_mut<TierInfo<CoinType>>(dd_addr);
+        reset_window(tier_info);
+        let cur_inflow = tier_info.window_inflow;
         let new_inflow = cur_inflow + amount;
-        let tiers = &mut tier.tiers;
+        let tiers = &tier_info.tiers;
         let number_of_tiers = Vector::length(tiers);
         assert(tier_index < number_of_tiers, Errors::invalid_argument(EINVALID_TIER_INDEX));
         let tier_upperbound: u64 = *Vector::borrow(tiers, tier_index);
         assert(new_inflow <= tier_upperbound, Errors::invalid_argument(EINVALID_AMOUNT_FOR_TIER));
-        tier.window_inflow = new_inflow;
+        tier_info.window_inflow = new_inflow;
+    }
+    spec fun validate_and_record_mint {
+        modifies global<TierInfo<CoinType>>(dd_addr);
```

Comment:
Conplete spec or drop obsolete modifies?

Reply:
My intention is to inline validate_and_record_mint; notice it is not opaque.  However, I still have to write the modifies spec because its caller demands it do so for sound modifies checking.